File: typescript-client-sdk/src/sdk/models/operations/findpets.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";
import * as utils from "../utils";
import { Type } from "class-transformer";


export class FindPetsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=tags" })
  tags?: string[];
}

export class FindPetsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: FindPetsQueryParams;

  @SpeakeasyMetadata()
  retries?: utils.RetryConfig;
}

export class FindPetsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  error?: shared.ErrorT;

  @SpeakeasyMetadata({ elemType: shared.Pet })
  pets?: shared.Pet[];

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-client-sdk/pkg/models/operations/findpets.go
package operations

import (
	"github.com/speakeasy-sdks/nolan-test/pkg/models/shared"
	"github.com/speakeasy-sdks/nolan-test/pkg/utils"
)

type FindPetsQueryParams struct {
	Limit *int     `queryParam:"style=form,explode=true,name=limit"`
	Tags  []string `queryParam:"style=form,explode=true,name=tags"`
}

type FindPetsRequest struct {
	QueryParams FindPetsQueryParams
	Retries     *utils.RetryConfig
}

type FindPetsResponse struct {
	ContentType string
	Error       *shared.Error
	Pets        []shared.Pet
	StatusCode  int
}
